packages/docusaurus/lib/client/pendingNavigation.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {Route, withRouter} from 'react-router-dom';
+import nprogress from 'nprogress';
+import 'nprogress/nprogress.css';
+import preload from './preload';
+
+let progressBarTimeout = null;
```

Comment:
This should be part of the component. If we put it here it means all instances of the component will share the same timeout. Even though there should only be one global spinner, it's still a better practice to put the timeout in the component.

```js
class ProgressBar extends React.Component {
  ...
  progressBarTimeout = null
}
```